src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/actions/RadiclePullAction.java

```diff
@@ -27,22 +27,20 @@ public void actionPerformed(@NotNull AnActionEvent e) {
     public void performAction(Project project) {
         var gitRepoManager = GitRepositoryManager.getInstance(project);
         var repos = gitRepoManager.getRepositories();
-        updateCountDown = new CountDownLatch(repos.size());
+        if (!BasicAction.isCliPathConfigured(project)) {
+            return ;
+        }
         ApplicationManager.getApplication().executeOnPooledThread(() -> {
-            if (BasicAction.isValidConfiguration(project)) {
-                repos.forEach(repo -> ApplicationManager.getApplication().executeOnPooledThread(()->{
-                    var pull = new RadPull(repo);
-                    new BasicAction(pull, repo, project, updateCountDown).perform();
-                }));
-
-                UpdateBackgroundTask ubt = new UpdateBackgroundTask(project, RadicleBundle.message("radPullProgressTitle"),
-                        updateCountDown, executingFlag);
-                new Thread(ubt::queue).start();
-            }
+            var radInitializedRepos = BasicAction.getInitializedReposWithNodeConfigured(repos, true);
+            updateCountDown = new CountDownLatch(radInitializedRepos.size());
```

Comment:
there should be a check whether we actually found any rad initialized repos, otherwise do not execute anything.